Tests/Unit/psgithub.tests.ps1

```diff
@@ -0,0 +1,69 @@
+#
+# This is a PowerShell Unit Test file.
+# You need a unit test framework such as Pester to run PowerShell Unit tests. 
+# You can download Pester from http://go.microsoft.com/fwlink/?LinkID=534084
+#
+
+$Global:ModuleRoot = Split-Path -Parent (Split-Path -Parent (Split-Path -Parent $Script:MyInvocation.MyCommand.Path))
+
+$OldLocation = Get-Location
+Set-Location -Path $ModuleRoot
+if (Get-Module PSGitHub -All)
+{
+    Get-Module PSGitHub -All | Remove-Module
+}
+
+Import-Module "$Global:ModuleRoot\PSGitHub.psd1" -Force -DisableNameChecking
+$Global:ArtifactPath = "$Global:ModuleRoot\Artifacts"
+$null = New-Item -Path "$Global:ArtifactPath" -ItemType Directory -Force -ErrorAction SilentlyContinue
+
+# Perform PS Script Analyzer tests on module code only
+$null = Set-PackageSource -Name PSGallery -Trusted -Force
+$null = Install-Module -Name 'PSScriptAnalyzer' -Confirm:$False
+Import-Module -Name 'PSScriptAnalyzer'
+
+Describe 'PSScriptAnalyzer' {
+    Context 'PSGitHub Module, Functions and TabCompleters' {
+        It 'Passes Invoke-ScriptAnalyzer' {
```

Comment:
The existing code is passing all "Error" level rules currently :+1: There are a few warnings (but they don't block build - just show up in the AppVeyor console log). The warnings are all about implementing "ShouldProcess" on the Set-*/New-*/Remove-* cmdlets.